cpp/tests/spatial/pip_compare.cu

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <time.h>
+#include <sys/time.h>
+#include <vector>
+#include <string>
+
+#include <utilities/error_utils.hpp>
+#include <cuspatial/pip.hpp>
+#include <gtest/gtest.h>
+#include "pip_util.h"
+
+#include <tests/utilities/cudf_test_utils.cuh>
+#include <tests/utilities/cudf_test_fixtures.h>
+
+using namespace cuspatial;
+
+template <typename T>
+struct PIPCompare : public GdfTest 
+{
+    T * h_point_x=nullptr, *h_point_y=nullptr;
```

Comment:
Should use cudf::column_wrapper and std::vector / device_vector rather than raw pointers. Initialization will be much simpler and safer.

Reply:
h_point_x and h_point_y are pointers that are used in cpu code, which does/can not use cudf.  They are also used to populate the data field of point_x and point_y columns, which shows an alternative approach to create gdf_columns than column_wrapper. Column_wrapper is used in pip_toy, which tests on hard coded toy data.

Reply:
Tests should not show alternative approaches. They should be succinct and easily maintainable. column_wrapper makes writing tests easier and less bug prone. Writing good tests typically is much more difficult and takes longer than writing the functionality itself, which is why testing utilities are important.

In general, raw pointers should be avoided when containers can be used because raw pointers often lead to memory leks.

Reply:
I agree. This is also the reason that I separated CPU code (used in _compare.cu only) and GPU code (used in both _test and _compare code). 
The CPU code does need access to raw pointers. I could pass gdf_columns to CPU code, but then the data need needs to be downloaded to CPU again, which is a unnecessary detour, in addition making the CPU code dependent on cudf which is unnecessary again.